src/main/java/picard/util/LiftoverUtils.java

```diff
@@ -293,17 +298,40 @@ public static VariantContext swapRefAlt(final VariantContext vc, final Collectio
         return swappedBuilder.make();
     }
 
+    /**
+     * Checks whether the reference allele in the provided variant context actually matches the reference sequence
+     *
+     * @param alleles list of alleles from which to find the reference allele
+     * @param referenceSequence the ref sequence
+     * @param start the start position of the actual indel
+     * @param end   the end position of the actual indel
+     * @return true if they match, false otherwise
+     */
+    protected static boolean referenceAlleleMatchesReferenceForIndel(final List<Allele> alleles,
+                                                                     final ReferenceSequence referenceSequence,
+                                                                     final int start,
+                                                                     final int end) {
+        final String refString = StringUtil.bytesToString(referenceSequence.getBases(), start - 1, end - start + 1);
+        final Allele refAllele = alleles.stream().filter(Allele::isReference).findAny().orElseThrow(() -> new IllegalStateException("Error: no reference allele was present"));
+        return (refString.equalsIgnoreCase(refAllele.getBaseString()));
+    }
 
     /**
      *    Normalizes and left aligns a {@link VariantContextBuilder}.
-     *    Note: this will modify the start/stop and alleles of this builder
+     *    Note: this will modify the start/stop and alleles of this builder.
+     *    Also note: if the reference allele does not match the reference sequence, this method does nothing
      *
      *    Based on Adrian Tan, Gonçalo R. Abecasis and Hyun Min Kang. (2015)
      *    Unified Representation of Genetic Variants. Bioinformatics.
      *
      */
     protected static void leftAlignVariant(final VariantContextBuilder builder, final int start, final int end, final List<Allele> alleles, final ReferenceSequence referenceSequence) {
 
+        //make sure that referenceAllele matches reference
+        if (!referenceAlleleMatchesReferenceForIndel(alleles, referenceSequence, start, end)) {
+            return;
```

Comment:
I'd prefer if you made it so that there's a requirement that the ref allele matches (in the javadoc) and threw an IllegalArgumentException if not true. this should work in the use-case here, since you check before going into this function, and it will protect against silent failures in the future.